sympy/printing/lambdarepr.py

```diff
@@ -163,6 +163,14 @@ def _print_Min(self, expr):
     def _print_Max(self, expr):
         return '{0}(({1}))'.format('amax', ','.join(self._print(i) for i in expr.args))
 
+    def _print_DotProduct(self, expr):
+        arg1, arg2 = expr.args
+        if arg1.shape[0] != 1:
```

Comment:
Can you explain the logic behind the transpositions here? According to numpy documentation for `dot`: 

> For 2-D arrays it is equivalent to matrix multiplication, and for 1-D arrays to inner product of vectors (without complex conjugation). For N dimensions it is a sum product over the last axis of a and the second-to-last of b

It's not obvious to me why we need the transpose; are the array objects we're operating on here always 2D?